wallet/node/Main.hs

```diff
@@ -0,0 +1,188 @@
+    notifierPlugins = ([ActionSpec $ \_ -> notifierPlugin], mempty)
+    allPlugins :: HasConfigurations => TVar NtpStatus -> ([WorkerSpec WalletWebMode], OutSpecs)
+    allPlugins ntpStatus =
+        mconcat [ convPlugins (plugins wArgs)
+                , walletProd wArgs ntpStatus
+                , acidCleanupWorker wArgs
+                , syncWalletWorker
+                , resubmitterPlugins
+                , notifierPlugins
+                ]
+
+syncWalletWorker :: HasConfigurations => ([WorkerSpec WalletWebMode], OutSpecs)
+syncWalletWorker =
+    first one $ worker mempty $ const $
+    modifyLoggerName (const "syncWalletWorker") $
+    (view (lensOf @SyncQueue) >>= processSyncRequest)
+
+acidCleanupWorker :: HasConfigurations => WalletArgs -> ([WorkerSpec WalletWebMode], OutSpecs)
+acidCleanupWorker WalletArgs{..} =
+    first one $ worker mempty $ const $
+    modifyLoggerName (const "acidcleanup") $
+    (askWalletDB >>= \db -> cleanupAcidStatePeriodically db walletAcidInterval)
+
+walletProd ::
+       ( HasConfigurations
+       , HasCompileInfo
+       )
+    => WalletArgs
+    -> TVar NtpStatus
+    -> ([WorkerSpec WalletWebMode], OutSpecs)
+walletProd WalletArgs {..} ntpStatus = first one $ worker walletServerOuts $ \diffusion -> do
+    logInfo $ sformat ("Production mode for API: "%build)
+        walletProductionApi
+    logInfo $ sformat ("Transaction submission disabled: "%build)
+        walletTxCreationDisabled
+    walletServeWebFull
+        diffusion
+        ntpStatus
+        walletDebug
+        walletAddress
+        (Just walletTLSParams)
+
+plugins ::
+    ( WorkMode ctx m
+    , HasNodeContext ctx
+    , HasConfigurations
+    , HasCompileInfo
+    ) => WalletArgs -> [m ()]
+plugins WalletArgs {..}
+    | enableWeb = [serveWeb webPort (Just walletTLSParams)]
+    | otherwise = []
+
+action :: HasCompileInfo => WalletNodeArgs -> Production ()
+action ntpConfig (WalletNodeArgs (cArgs@CommonNodeArgs{..}) (wArgs@WalletArgs{..})) =
+    withConfigurations conf $ \ntpConfig -> do
+        CLI.printInfoOnStart cArgs
+        logInfo $ "Wallet is enabled!"
+        currentParams <- getNodeParams loggerName cArgs nodeArgs
+
+        let vssSK = fromJust $ npUserSecret currentParams ^. usVss
+        let sscParams = CLI.gtSscParams cArgs vssSK (npBehaviorConfig currentParams)
+
+        actionWithWallet sscParams currentParams ntpConfig wArgs
+  where
+    nodeArgs :: NodeArgs
+    nodeArgs = NodeArgs { behaviorConfigPath = Nothing }
+
+    conf :: ConfigurationOptions
+    conf = CLI.configurationOptions $ CLI.commonArgs cArgs
+
+
+main :: IO ()
+main = withCompileInfo $(retrieveCompileTimeInfo) $ do
+    args <- getWalletNodeOptions
+    let loggingParams = CLI.loggingParams loggerName (wnaCommonNodeArgs args)
+    loggerBracket loggingParams . logException "node" . runProduction $ do
+        logInfo "[Attention] Software is built with wallet part"
+        action args
```

Comment:
I thought we deleted this file, or moved it somewhere else.

Reply:
Yeah, I was wondering why it was in this part of the tree.

Reply:
Fixed.